### PR TITLE
[DPE-5032] Release 3.5.1-ubuntu1

### PR DESCRIPTION
**Pull request by deusebio**

### CODE REVIEW

**Review by theoctober19th — APPROVED** [review]

**Comment by deusebio** [comment]
The track 3.5 on snapstore has been created, see [here](https://forum.snapcraft.io/t/spark-client-snap-track-name/35619/13), so we should be able to publish when merging this PR

File: tests/integration/ie-tests.sh [hunk 1]
```diff
@@ -424,13 +424,13 @@ run_spark_submit_custom_certificate(){
...
-      --conf spark.kubernetes.container.image=ghcr.io/canonical/charmed-spark:3.4.2-22.04_edge
+      --conf spark.kubernetes.container.image=ghcr.io/canonical/charmed-spark:3.5.1-22.04_edge
   
   echo "Print current config."
   spark-client.service-account-registry get-config --username hello
 
   echo "Run Spark job"
-  spark-client.spark-submit --username hello -v --conf spark.hadoop.fs.s3a.connection.ssl.enabled=true --conf spark.kubernetes.container.image=ghcr.io/canonical/charmed-spark:3.4.2-22.04_edge --conf spark.kubernetes.executor.request.cores=0.1 --files="./tests/integration/resources/example.txt" --class org.apache.spark.examples.SparkPi local:///opt/spark/examples/jars/spark-examples_2.12-3.4.2.jar 100
+  spark-client.spark-submit --username hello -v --conf spark.hadoop.fs.s3a.connection.ssl.enabled=true --conf spark.kubernetes.container.image=ghcr.io/canonical/charmed-spark:3.5.1-22.04_edge --conf spark.kubernetes.executor.request.cores=0.1 --files="./tests/integration/resources/example.txt" --class org.apache.spark.examples.SparkPi local:///opt/spark/examples/jars/spark-examples_2.12-3.5.1.jar 100
```
Comment (welpaolo) [inline tests/integration/ie-tests.sh]:
```suggestion
  spark-client.spark-submit --username hello -v --conf spark.hadoop.fs.s3a.connection.ssl.enabled=true --conf spark.kubernetes.container.image=ghcr.io/canonical/charmed-spark:3.5.1-22.04_edge --conf spark.kubernetes.executor.request.cores=0.1 --files="./tests/integration/resources/example.txt" --class org.apache.spark.examples.SparkPi $SPARK_EXAMPLES_JAR_NAME 100
```

Comment (welpaolo) [reply tests/integration/ie-tests.sh]: nitpick

**Review by welpaolo — APPROVED** [review]
LGTM

File: tests/integration/ie-tests.sh (hunk 1 above)
Comment (deusebio) [reply tests/integration/ie-tests.sh]: Sure! nice!

**Review by welpaolo — APPROVED** [review]
LGTM

**Pull request merged**